src/service/state.service.ts

```diff
@@ -33,27 +31,36 @@ export class StateService {
     return contents[key];
   }
 
-  public async safeRead(key: string) {
-    if (await this.has(key)) {
-      return this.read(key);
-    }
-    return null;
-  }
-
   public async has(key: string) {
     const contents = await this.readFile();
     return contents[key] !== undefined;
   }
 
-  private async readFile() {
+  public async readFile() {
     if (await checkFileExists(this.storeFilePath)) {
       return JSON.parse(String(await readFile(this.storeFilePath)));
     }
 
     return {};
   }
 
-  private async writeFile(contents) {
-    await writeFile(this.storeFilePath, JSON.stringify(contents, null, 2), {mode: 0o660});
+  public async writeFile(contents) {
+    while (await this.isLocked()) {
+      await this.isLocked();
+    }
+    this.lock = true;
+    await writeFile(this.storeFilePath, JSON.stringify(contents, null, 2), {mode: 0o660})
+          .finally(() => this.lock = false);
   }
+
+  private isLocked = async(): Promise<boolean> => {
+    if (this.lock) {
+      await this.lockSleep(this.lockWaitMs);
```

Comment:
Maybe this should be moved to line 49? Instead of await this.isLocked();?

Reply:
[02c5857](https://github.com/ambrosus/ambrosus-node-extended/pull/175/commits/02c585706e366f96f3be00236418545a18354a4d) simplified